source/Symbol/SwiftASTContext.cpp

```diff
@@ -3472,9 +3472,16 @@ bool SwiftASTContext::LoadOneImage(Process &process, FileSpec &link_lib_spec,
 
 static std::vector<std::string>
 GetLibrarySearchPaths(const swift::SearchPathOptions &search_path_opts) {
-  std::vector<std::string> paths(search_path_opts.LibrarySearchPaths.begin(),
-                                 search_path_opts.LibrarySearchPaths.end());
+  // The order in which we look up the libraries is important. The REPL
+  // dlopen()s libswiftCore, and gives precedence to the just built standard
+  // library instead of the one in the OS. When we type `import Foundation`,
+  // we want to make sure we end up loading the correct library, i.e. the
+  // one sitting next to the stdlib we just built, and then fall back to the
+  // one in the OS if that's not available.
+  std::vector<std::string> paths;
   paths.push_back(search_path_opts.RuntimeLibraryPath);
```

Comment:
I’m curious how we mirror the situation of the different rpaths for builds and installs. What’s the value in the later scenario? Is it empty?

Reply:
Are you talking about the value of `RuntimeLibraryPath`? If so, yes, it's empty.

Reply:
Alright, that's what I expected.

Reply:
For Swift 5.1, we are moving to have Apple's toolchains not contain the Swift dylibs so the only case where this code will find libraries in the toolchain is when doing development or using a downloaded Swift.org toolchain. In that case, you should set DYLD_LIBRARY_PATH to prefer the toolchain libraries. If you don't set DYLD_LIBRARY_PATH, it may still not work because other dylibs may have dependencies on /usr/lib/swift and you'll still end up with multiple versions of the same library getting loaded. I'm not sure about the context of this code, but if that's what it is doing, then you should be ok.